tests/test_2512_record_array_carry.py

```diff
@@ -0,0 +1,67 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+import numpy as np
+
+import awkward as ak
+
+
+def test():
+    def _reduce_max_masked(array, mask):
+        assert mask
+        j = ak.from_regular(
+            ak.argmax(array["1"], axis=1, keepdims=True, mask_identity=True)
+        )
+        return ak.flatten(array[j], axis=1)
```

Comment:
I'm being "lazy" by avoiding implementing an identity. The identity can easily be obtained using `Form.length_zero_array()` and `fill_none`, but it's not necessary here.